### PR TITLE
InventoryExport APIs do not correctly support sales channels

### PR DESCRIPTION
**Pull request by bartekszymanski**
### Description (*)
By default MSI installations use Website as a Sales Channel, however, system should support customisations changing this behavior and assigning sales channel to something else.

### Fixed Issues (if relevant)

1. magento-engcom/msi#2211: InventoryExport APIs do not correctly support sales channels

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by ishakhsuvarov** [comment]
Hi @bartekszymanski 
Thanks for the PR! Could you please take a look at the api-functional test failure? I assume you have used that magical number when testing, but want to double check just in case.

**Comment by smoskaluk** [comment]
Hi @bartekszymanski , I got error trying to get export, may you please recheck your PR

image

**Comment by ishakhsuvarov** [comment]
@smoskaluk This PR changed both REST API endpoints.

Instead of 
`/V1/inventory/export-stock-salable-qty?searchCriteria`
it's now
`/V1/inventory/export-stock-salable-qty/:salesChannelCode?searchCriteria`

So `salesChannelCode` has to be added as well to make it work. Please try again

File: app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyInterface.php [hunk 1]
```diff
@@ -17,11 +17,11 @@ interface ExportStockSalableQtyInterface
...
      *
      * @param \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
-     * @param string $websiteCode
+     * @param $salesChannelCode $salesChannelCode
      * @return \Magento\InventoryExportStockApi\Api\Data\ExportStockSalableQtySearchResultInterface
      */
-    public function execute(
-        \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria,
-        string $websiteCode
+    public function getList(
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyInterface.php]: I'd stick to `execute` method name as it's already became a convention.

File: app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyBySalesChannelInterface.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryExportStockApi\Api;
+
+interface ExportStockSalableQtyBySalesChannelInterface
+{
+    public function getList(
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyBySalesChannelInterface.php]: I'd stick to `execute` method name as it's already became a convention.

File: app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyBySalesChannelInterface.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryExportStockApi\Api;
+
+interface ExportStockSalableQtyBySalesChannelInterface
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyBySalesChannelInterface.php]: Please add description

File: app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyBySalesChannelInterface.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyBySalesChannelInterface.php]: Currently to me it seems like we are pretty much duplicating the `ExportStockSalabeQtyInterface`. 

The only case I could think of now would be the need to tune export based on the Sales Channel extension attributes, but I can hardly come up with the real scenario

File: app/code/Magento/InventoryExportStockApi/Api/ExportStockIndexDataBySalesChannelInterface.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?php
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryExportStockApi/Api/ExportStockIndexDataBySalesChannelInterface.php]: Do we really need to proxy through one more Interface?

File: app/code/Magento/InventoryExportStockApi/Api/ExportStockIndexDataBySalesChannelInterface.php [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryExportStockApi\Api;
+
+interface ExportStockIndexDataBySalesChannelInterface
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryExportStockApi/Api/ExportStockIndexDataBySalesChannelInterface.php]: Description would be helpful

File: app/code/Magento/InventoryExportStockApi/Test/Api/ExportStockIndexDataTest.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+use Magento\TestFramework\TestCase\WebapiAbstract;
+use Magento\Framework\Webapi\Rest\Request;
+
+class ExportStockIndexDataTest extends WebapiAbstract
+{
+    const API_PATH = '/V1/inventory/dump-stock-index-data';
+    const SERVICE_NAME = 'inventoryExportStockApiExportStockIndexDataV1';
+
+    const EXPORT_PRODUCT_COUNT = 1206;
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryExportStockApi/Test/Api/ExportStockIndexDataTest.php]: 1206 seems to be a magical number used during testing. Should we adjust it to 6 or even calculate dynamically?

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]
Interfaces structure remains unclear to me, I'd discuss it additionally if there are any use cases which require additional APIs

**Comment by bartekszymanski** [comment]
i will do it all next week I'm still in USA after Imagine traveling a lot

**Pull request merged**